src/service.rs

```diff
@@ -213,12 +221,25 @@ struct ActiveRequest {
     pub callback: Option<CallbackResponse>,
 }
 
+#[derive(Debug)]
+#[allow(dead_code)]
```

Comment:
Why do we have `allow(dead_code) ` here? Its used to construct the response right?

Reply:
```
error: fields `enr_seq`, `ip`, and `port` are never read
   --> src\service.rs:227:5
    |
225 | pub struct Pong {
    |            ---- fields in this struct
226 |     /// The current ENR sequence number of the responder.
227 |     enr_seq: u64,
    |     ^^^^^^^
228 |     /// Our external IP address as observed by the responder.
229 |     ip: IpAddr,
    |     ^^
230 |     /// Our external UDP port as observed by the responder.
231 |     port: u16,
    |     ^^^^
    |
    = note: `Pong` has a derived impl for the trait `Debug`, but this is intentionally ignored during dead code analysis
    = note: `-D dead-code` implied by `-D warnings`

error: could not compile `discv5` due to previous error 
```

```let response = Pong { enr_seq, ip, port };```

It is used to construct the response, but when I remove that clippy complains there is dead code for some reason?

Reply:
Oh I just realized I should make them public, I wonder why I didn't think of that.

Reply:
Yeah if they are not public, then the data would be unusable in the app?

I assume then this is not tested in the app side where you need it?

Reply:
It isn't tested yet no. When I make a pr for trin adding support for this command I will have to write a test for it there. I was planning to do that after this was merged. I still need to ask the others if we only pull crates from releases now or also commits. Cause if we only do it from releases now then I might have to wait a while.

Reply:
Sure. You could just do quick tests against the branch. That would have found things like not being able to access the data you want and us having to do another PR to correct it later. 

At the moment I think this is fine. I was thinking of getting your two PRs in and doing a release. I've also updated to the latest ENR.

Reply:
I will test that tomorrow after my math final then and let you know :)